Kommunicate/Classes/KMConversationViewController.swift

```diff
@@ -12,32 +12,71 @@ import ApplozicSwift
 /// Before pushing this view Controller. Use this
 /// navigationItem.backBarButtonItem = UIBarButtonItem(customView: UIView())
 open class KMConversationViewController: ALKConversationViewController {
-    
+
     public var kmConversationViewConfiguration: KMConversationViewConfiguration!
-    
+
     lazy var customNavigationView = ConversationVCNavBar(navigationBarBackgroundColor: self.configuration.navigationBarBackgroundColor, delegate: self, configuration: kmConversationViewConfiguration)
-    
+
+    let awayMessageView = AwayMessageView(frame: CGRect.zero)
+
     lazy var channelKey = self.viewModel.channelKey
-    
+    private let awayMessageheight = 80.0
+
     override open func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         setupNavigation()
-        
+
         // Fetch Assignee details every time view is launched.
         updateAssigneeDetails()
+        messageStatus()
     }
-    
+
     open override func viewDidLoad() {
         super.viewDidLoad()
+
         checkPlanAndShowSuspensionScreen()
+        addAwayMessageConstraints()
+        showAwayMessage(false)
+    }
+
+    open override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        awayMessageView.drawDottedLines()
+    }
+
+    override open func newMessagesAdded() {
+        super.newMessagesAdded()
+
+        showAwayMessage(false)
+    }
+
+    func addAwayMessageConstraints() {
+        chatBar.headerView.addViewsForAutolayout(views: [awayMessageView])
+        awayMessageView.layout {
+            $0.leading == chatBar.headerView.leadingAnchor
+            $0.trailing == chatBar.headerView.trailingAnchor
+            $0.bottom == chatBar.headerView.bottomAnchor
+            $0.height == chatBar.headerView.heightAnchor
+        }
+    }
+
+    func messageStatus() {
+        guard let channelKey = viewModel.channelKey else { return }
+        KMConversationService().statusForUser(channelKey, completion: { message in
```

Comment:
Can we use a global variable here it would be difficult to test this method otherwise? Also, write few test cases for this method.